### PR TITLE
add getCurrentConnectionInfo helper

### PR DESCRIPTION
**Pull request by SpiralP**
also add jest!

### CODE REVIEW

File: src/getCurrentConnectionInfo.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+/** returns info on the current Portal/Cloud connection */
+export const getCurrentConnectionInfo = (
+  window: Window
+): ConnectionInfo | null => {
+  const urlPathRegex = /\/u\/(?<index>[0-9]+)\/(?<product>.+?)\//;
+  const currentMatch = window.location.pathname.match(urlPathRegex);
+  if (
+    currentMatch &&
+    currentMatch.groups &&
```
Comment (rmyers) [inline src/getCurrentConnectionInfo.ts:16]: I think if there is a match the `groups` should not be null, and `groups.index` should always be set too

File: src/cookies/__tests__/getCurrentPortalUrlInfo.test.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    expect(info?.connectionIndex).toBe(0);
+    expect(info?.product).toBe('explorer');
+  });
+
+  it('returns null for invalid urls', () => {
+    const window = { ...windowFixture };
+
+    expect(getCurrentConnectionInfo(window)).toBeTruthy();
+
+    window.location.href = 'http://localhost:8080/u/0/';
```
Comment (rmyers) [inline src/cookies/__tests__/getCurrentPortalUrlInfo.test.ts:31]: Seems like this should be a valid url (for portal itself). So this should match correctly. However a route like `http://foo/notu/0/` should not match even though there is a `u/number/`

Comment (SpiralP) [reply src/cookies/__tests__/getCurrentPortalUrlInfo.test.ts:31]: so you're saying:
* add a ^ starts with requirement like `^/u/...`
* and make the `/u/0/<product>/` product part optional?

File: src/getCurrentConnectionInfo.ts (hunk 1 above)
Comment (SpiralP) [reply src/getCurrentConnectionInfo.ts]: typescript thinks groups can be undefined, but i can use the ?. operator:
[`7004d97` (#8)](https://github.com/stardog-union/portal-sdk/pull/8/commits/7004d97663032ef7dae6f804a2521c85eedba473)

File: src/cookies/__tests__/getCurrentPortalUrlInfo.test.ts (hunk 2 above)
Comment (SpiralP) [reply src/cookies/__tests__/getCurrentPortalUrlInfo.test.ts]: something like this?
[`dd58d7c` (#8)](https://github.com/stardog-union/portal-sdk/pull/8/commits/dd58d7cdb3c1e4af9c4975267879eaf63c9b6a19)

Comment (SpiralP) [reply src/cookies/__tests__/getCurrentPortalUrlInfo.test.ts]: alright, here's the regex update:
[`18ca9ae` (#8)](https://github.com/stardog-union/portal-sdk/pull/8/commits/18ca9ae7461db2c34a180ed48caecc7ce51c9d75)

File: src/getCurrentConnectionInfo.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  /** an optional string like "designer" | "explorer" | "studio" */
+  product?: string;
+};
+
+/** returns info on the current Portal/Cloud connection */
+export const getCurrentConnectionInfo = (
+  window: Window
+): ConnectionInfo | null => {
+  const urlPathRegex =
+    /^\/u\/(?<connectionIndex>[0-9]+)\/(?<productWithSlash>.+?\/)?/;
```
Comment (rmyers) [inline src/getCurrentConnectionInfo.ts:13]: You can simplify this regex to this:

```
const urlPathRegex = /^\/u\/(?<connectionIndex>[0-9]+)\/(?<product>[^/]+)?/;
```

Screen Shot 2022-03-02 at 4 51 18 PM

Then you don't need to remove the slash instead just include all chars except `/` you could also change this to `/(?<product>[^/]*)/` which will match 0 or more non-slashes, then product will be an empty string

**Review by rmyers — CHANGES_REQUESTED** [review]
minor change to the regex and then it is good

File: src/getCurrentConnectionInfo.ts (hunk 3 above)
Comment (SpiralP) [reply src/getCurrentConnectionInfo.ts]: [`65fc956` (#8)](https://github.com/stardog-union/portal-sdk/pull/8/commits/65fc95676eef30aff1fa4ee976acb2a28be37ad6)

**Review by rmyers — APPROVED** [review]

**Pull request merged**